src/OS/QMidi_Haiku.cpp

```diff
@@ -79,10 +83,183 @@ void QMidiOut::sendMsg(qint32 msg)
 	if (!fConnected)
 		return;
 
+	size_t bufferLength = 3;
 	char buf[3];
 	buf[0] = msg & 0xFF;
 	buf[1] = (msg >> 8) & 0xFF;
 	buf[2] = (msg >> 16) & 0xFF;
 
-	fMidiPtrs->midiOutLocProd->SprayData((void*)&buf, 3, true);
+	if (buf[0] == '\xC0' || buf[0] == '\xD0') {
+		// workaround for Haiku bug #15562
```

Comment:
So, on the proposed patch for that bug (https://review.haiku-os.org/c/haiku/+/2048), it was determined that this packet is actually short 1 byte. The driver should not crash, but that would indicate that this is somehow not doing the right thing?

Reply:
hmm, yeah... I guess the right thing to do here is to not use `SprayData` directly but rather the other family of `Spray*` methods depending on the message that needs to be sent.  To me it seems that the nicest way to handle that would be to pass the `QMidiEvent`s down to the `sendMsg` methods and then "just" perform a `switch (event.type()) { ... }` which calls the corresponding method in the different `case`s.  

However judging by the docs of some other MIDI APIs out there which are not yet implemented in QMidi (e.g. CoreAudio: [MIDISend](https://developer.apple.com/documentation/coremidi/1495289-midisend?language=objc), [struct MIDIPacket](https://developer.apple.com/documentation/coremidi/midipacket?language=objc)) it seems that only the Haiku implementation would benefit from passing a `QMidiEvent` to the `sendMsg` method; most other APIs just want to deal with the raw MIDI bytes instead (I mean... just look at the [Swift initialiser for the MIDIPacket struct](https://developer.apple.com/documentation/coremidi/midipacket?language=swift) 🤯).  
I reckon the sane thing to do here is to just do a `switch (buf[0])` here then (and call `SpraySystemExclusive` in the `sendSysEx` method).

Reply:
Sounds good.

Reply:
That's done now :tada: